src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -151,17 +151,73 @@ void ClusterTaskManager::TasksUnblocked(const std::vector<TaskID> ready_ids) {
   }
 }
 
+void ClusterTaskManager::TaskFinished(const TaskID &task_id) {
+  auto iter = running_tasks_.find(task_id);
+  if (iter != running_tasks_.end()) {
+    running_tasks_.erase(iter);
+  } else {
+    // This shouldn't happen (unless the worker resends a ReturnWorker request?).
+    RAY_LOG(DEBUG) << "Received TaskDone for unknown task.";
+  }
+}
+
+bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
+  for (auto iter = tasks_to_schedule_.begin(); iter != tasks_to_schedule_.end(); iter++) {
+    if (std::get<0>(*iter).GetTaskSpecification().TaskId() == task_id) {
```

Comment:
I believe you can also do `iter->first`.

Reply:
Does `std::tuple` have a `first`?

Reply:
Ah thought this was a pair.